### DISCUSSION TITLE
Freetds installation error in termux

### DISCUSSION THREAD

**Discussion by nafish-akib** [FreeTDS/freetds#516] [Q&A] [answered]
![Screenshot_20231202-010208_Termux](https://github.com/FreeTDS/freetds/assets/112865518/8ef36e39-3b1e-4339-841e-bd076c28878d)
make[3]: Entering directory '/data/data/com.termux/files/home/freetds/src/utils'
  CC       md4.lo
  CC       md5.lo
  CC       des.lo
  CC       hmac_md5.lo
  CC       dlist.lo
  CC       getpassarg.lo
  CC       sleep.lo
  CC       tds_cond.lo
tds_cond.c:215:9: error: call to undeclared function 'pthread_cond_timedwait_relative_np'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  215 |         return pthread_cond_timedwait_relative_np(cond, mtx, &ts);
      |                ^
tds_cond.c:215:9: note: did you mean 'pthread_cond_timedwait_monotonic_np'?
/data/data/com.termux/files/usr/include/pthread.h:164:5: note: 'pthread_cond_timedwait_monotonic_np' declared here
  164 | int pthread_cond_timedwait_monotonic_np(pthread_cond_t* _Nonnull __cond, pthread_mutex_t* _Nonnull __mutex,
      |     ^
1 error generated.
make[3]: *** [Makefile:534: tds_cond.lo] Error 1
make[3]: Leaving directory '/data/data/com.termux/files/home/freetds/src/utils'
make[2]: *** [Makefile:554: all-recursive] Error 1
make[2]: Leaving directory '/data/data/com.termux/files/home/freetds/src/utils'
make[1]: *** [Makefile:425: all-recursive] Error 1
make[1]: Leaving directory '/data/data/com.termux/files/home/freetds/src'
make: *** [Makefile:466: all-recursive] Error 1

I can't fix it. Please help.

**Comment by freddy77** [comment]
Apparently during configuration `pthread_cond_timedwait_relative_np` function was detected but your headers do not define it correctly. See logs in `config.log`. You can manually remove `HAVE_PTHREAD_COND_TIMEDWAIT_RELATIVE_NP` definition from `include/config.h` after configure.

**Comment by freddy77** [comment]
It should work with master without changes now. Can you try it?

**Reply by nafish-akib** [reply]
I will try

**Reply by nafish-akib** [reply]
![Screenshot_20231204-011207_Termux](https://github.com/FreeTDS/freetds/assets/112865518/5d816a0d-6ff3-4adb-bb26-80735fe1cd18)


tds_cond.c:226:16: error: use of undeclared identifier 'tv'
  226 |         gettimeofday(&tv, NULL);                         |                       ^
tds_cond.c:227:14: error: use of undeclared identifier 'tv'  227 |         ts.tv_sec = tv.tv_sec;
      |                     ^                              tds_cond.c:228:15: error: use of undeclared identifier 'tv'
  228 |         ts.tv_nsec = tv.tv_usec * 1000u;                 |                      ^
3 errors generated.                                        make[3]: *** [Makefile:534: tds_cond.lo] Error 1

**Comment by freddy77** [comment, marked as answer]
Okay, added a new commit, I did some more testing trying to simulate the issue, it should work now.

**Comment by nafish-akib** [comment]
It's now working perfectly..
